src/arduino.c

```diff
@@ -37,6 +37,45 @@
 #include "stk500.h"
 #include "arduino.h"
 
+static int arduino_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
+  const char *extended_param;
+  int attempts;
+  int rv = 0;
+
+  for (LNODEID ln = lfirst(extparms); ln; ln = lnext(ln)) {
+    extended_param = ldata(ln);
+
+    if (sscanf(extended_param, "attempts=%2d", &attempts) == 1) {
```

Comment:
`%2d` is a weird format, as it *only* reads 2 digits of the given number, but the user *doesn't know* and *it's hard to document*, so the user will be puzzled why `-xattempts=123` set the number of attempts to 12. Why not use `%i`? That allows hex numbers (`0x12`) or octal numbers (`077`), too. Normally one should also check that the number makes sense (must be positive here). Try `-xattempts=-1` and you'll see what I mean.

Reply:
Sure! `%2d` is just copy-paste from stk500.c, but you have a point!

Reply:
Thanks. This means the `stk500.c` code also needs changing (as that too is wrong)

Reply:
> stk500.c code also needs changing

You already did! Apologies for stating the obvious.